c10/util/intrusive_ptr.h

```diff
@@ -215,12 +247,17 @@ inline uint32_t atomic_weakcount_increment(std::atomic<uint32_t>& weakcount) {
 // final decrement has to load-acquire, either directly or with the help of
 // fences. But it's easiest just to have all decrements be acq-rel. And it turns
 // out, on modern architectures and chips, it's also fastest.
-inline uint32_t atomic_refcount_decrement(std::atomic<uint32_t>& refcount) {
-  return refcount.fetch_sub(1, std::memory_order_acq_rel) - 1;
+inline uint64_t atomic_combined_refcount_decrement(
+    std::atomic<uint64_t>& combined_refcount,
+    uint64_t dec) {
+  return combined_refcount.fetch_sub(dec, std::memory_order_acq_rel) - dec;
 }
 
-inline uint32_t atomic_weakcount_decrement(std::atomic<uint32_t>& weakcount) {
-  return weakcount.fetch_sub(1, std::memory_order_acq_rel) - 1;
+inline uint32_t atomic_weakcount_decrement(
+    std::atomic<uint64_t>& combined_refcount) {
+  return (uint32_t)(atomic_combined_refcount_decrement(
```

Comment:
Should this be static cast or reinterpret cast?

Reply:
Converting uint64_t to uint32_t should always use static_cast.